src/login/logind-user.c

```diff
@@ -612,9 +612,9 @@ int user_finalize(User *u) {
         if (k < 0)
                 r = k;
 
-        /* Clean SysV + POSIX IPC objects */
-        if (u->manager->remove_ipc) {
-                k = clean_ipc(u->uid);
+        /* Clean SysV + POSIX IPC objects, but only if this is not a system user. */
+        if (u->manager->remove_ipc && u->uid > SYSTEM_UID_MAX) {
```

Comment:
This check was always there, but previously inside of clean_ipc(). As we use this very same function now in both logind (to clean-up IPC objects of logged-in users) as well as from PID 1 (to clean-up IPC objects of system services) it was generalized a bit, and I removed the system UID check from it.

Background: unfortunately in many setups cronjobs actually register full login sessions, even if they run code that belongs to system services. removing the IPC objects of system users is  problematic from logind as this might mean we potentially drop IPC objects from system services we really shouldn't be dropping. After all RemoveIPC= in logind is opt-out while the new RemoveIPC= in system services is opt-in. Thus we need to be more careful in logind than in systemd's service manager. In the latter we only do IPC clean-ups on explicit configuration. In the former we do it for every session logged in.

I'll add a longer comment about this to the sources.